redis/client.py

```diff
@@ -310,6 +320,12 @@ def __init__(
         else:
             self.response_callbacks.update(_RedisCallbacksRESP2)
 
+        self.client_cache = client_cache
+        self.cache_blacklist = cache_blacklist
```

Comment:
Shouldn't all these values be set only if client_cache is true? I'd assume blacklist/whitelist to be empty, when False, we can save the memory, at the very least.